### PR TITLE
add use lanes menu

### PR DESCRIPTION
**Pull request by luvkapur**
## Proposed Changes

- Update Lane Overview  to render a `UseBox` to prompt users on how to import components from a lane

**Checked out to the Lane**

image

**Viewing a Lane**

image

### CODE REVIEW

**Comment by KutnerUri** [comment]
@luvkapur - what are the command snippets you give the user?

File: scopes/lanes/ui/menu/use-lanes-menu.module.scss [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  }
+  > div:nth-child(2) {
+    margin-bottom: 24px;
+  }
+}
+
+.drawerTitle {
+  display: flex;
+  align-items: center;
+  > span {
```
Comment (KutnerUri) [inline scopes/lanes/ui/menu/use-lanes-menu.module.scss:32]: same

File: scopes/lanes/ui/menu/use-lanes-menu.module.scss [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  padding: 16px 24px;
+  color: var(--bit-text-color, #414141);
+}
+
+.title {
+  display: flex;
+  align-items: center;
+  font-size: var(--bit-p-xs, 14px);
+  font-weight: bold;
+  > span {
```
Comment (KutnerUri) [inline scopes/lanes/ui/menu/use-lanes-menu.module.scss:11]: same

File: scopes/lanes/ui/hooks/lanes-query/lanes-query.ts [hunk 3]
```diff
@@ -24,13 +24,16 @@ const GET_LANES = gql`
...
+> {
   const { data, ...rest } = useDataQuery<LanesQueryResult>(GET_LANES);
   const { scope, loading } = useScopeQuery();
-  const lanes = data && LanesModel.from(data, scope);
   return {
     ...rest,
     loading: rest.loading || !!loading,
-    lanes,
+    lanes: data && LanesModel.from(data, scope),
+    checkedoutLane: data?.lanes?.getCurrentLaneName,
```
Comment (KutnerUri) [inline scopes/lanes/ui/hooks/lanes-query/lanes-query.ts:37]: that's.. really confusing. you are checking if checkoutLane !=== current lane in a different file

File: scopes/lanes/ui/menu/use-lanes-menu.tsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+export type LaneImportContentProps = {
+  currentLane: LaneModel;
+  switchedOutToCurrentLane: boolean;
+} & React.HTMLAttributes<HTMLDivElement>;
+
+export function UseLaneMenu() {
+  const lanesContext = useLanesContext();
+  if (!lanesContext?.currentLane) return null;
+  const { currentLane, checkedoutLane } = lanesContext;
+  const switchedOutToCurrentLane = currentLane.id === checkedoutLane?.id;
```
[the remark this answers is not in the document]
Comment (luvkapur) [reply scopes/lanes/ui/menu/use-lanes-menu.tsx:20]: alternative,

`selectedLane`
`checkedoutLane`

File: scopes/lanes/ui/hooks/lanes-query/lanes-query.ts (hunk 3 above)
Comment (luvkapur) [reply scopes/lanes/ui/hooks/lanes-query/lanes-query.ts:37]: From the CLI's perspective (`getCurrentLane`) exposed from the endpoint represents the currently checked out lane
which is always `main` on a scope but locally can be whatever the lane you are currently switched to

In the UI, the user can view a lane without having to switch to it. 

When you are switched out locally to the lane you are currently viewing on the UI,
the command to update your components from remote is different 
v/s
when you are just viewing a lane you are not switched to

File: scopes/lanes/ui/menu/use-lanes-menu.tsx [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    return (
+      <div className={styles.importContent}>
+        <div>Import everything from {currentLane.name}</div>
+        <CopyBox>{`bit merge`}</CopyBox>
+      </div>
+    );
+  }
+  return (
+    <div className={styles.importContent}>
+      <div>Switch and Import everything from {currentLane.name}</div>
```
[the remark this answers is not in the document]
Comment (luvkapur) [reply scopes/lanes/ui/menu/use-lanes-menu.tsx:56]: Except for docs lets discuss what more information is necessary here

File: scopes/lanes/ui/menu/use-lanes-menu.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (KutnerUri) [reply scopes/lanes/ui/menu/use-lanes-menu.tsx:20]: we talked about this naming:
`selected` => `viewed`
`checkedout` => 'switchedIn` (bit term) / ~~`active`~~ (might conflict with "inactive" lanes) / `current` (consistent with cli)

File: scopes/lanes/ui/menu/use-lanes-menu.module.scss [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+.importContent {
+  > div {
+    margin: 12px 0;
+  }
+  > div:nth-child(2) {
+    margin-bottom: 24px;
+  }
+}
```
[the remark this answers is not in the document]
Comment (luvkapur) [reply scopes/lanes/ui/menu/use-lanes-menu.module.scss]: will refactor `scope-menu` in a separate branch. With the refactor of current and viewed lane, this PR is already big. Want to reduce the scope of testing as well.

**Pull request merged**